src/nvim/api/win_config.c

```diff
@@ -240,6 +247,49 @@ void nvim_win_set_config(Window window, Dict(float_config) *config, Error *err)
   }
 }
 
+Dictionary config_put_bordertext(Dictionary config, FloatConfig *fconfig, BorderTextType bordertext_type)
+{
+  VirtText chunks;
+  AlignTextPos align;
+  char* field_name;
+  char* field_pos_name;
+  if (bordertext_type == kBorderTextTitle) {
+    chunks = fconfig->title_chunks;
+    align = fconfig->title_pos;
+    field_name = "title";
+    field_pos_name = "title_pos";
+  } else if (bordertext_type == kBorderTextFooter) {
+    chunks = fconfig->footer_chunks;
+    align = fconfig->footer_pos;
+    field_name = "footer";
+    field_pos_name = "footer_pos";
+  }
```

Comment:
These two `if` statements cover all variants of `BorderTextType`, but if a new variant is ever added and used you will end up with a read of an uninitialized variable (`field_name` and `field_pos_name`).

Suggestion: either (1) use a `switch` statement (so the compiler will warn you if a new unhandled variant is ever added), or (2) initialize `field_name` and `field_pos_name` to `NULL`, and below this `if` statement add `assert(field_name != NULL)`.